tests/unit/test_settings.py

```diff
@@ -0,0 +1,345 @@
+        self.assertRaises(
+            AttributeError,
+            lambda: "compress_rom" in test_settings.__getattribute__("compress_rom"),
+        )
+
+    def test_old_compress_setting_compressed(self):
+        test_settings = Settings.Settings({"compress_rom": "True"})
+
+        self.assertFalse(test_settings.create_patch_file)
+        self.assertTrue(test_settings.create_compressed_rom)
+        self.assertFalse(test_settings.create_uncompressed_rom)
+        self.assertRaises(
+            AttributeError,
+            lambda: "compress_rom" in test_settings.__getattribute__("compress_rom"),
+        )
+
+    def test_old_compress_setting_uncompressed(self):
+        test_settings = Settings.Settings({"compress_rom": "False"})
+
+        self.assertFalse(test_settings.create_patch_file)
+        self.assertFalse(test_settings.create_compressed_rom)
+        self.assertTrue(test_settings.create_uncompressed_rom)
+        self.assertRaises(
+            AttributeError,
+            lambda: "compress_rom" in test_settings.__getattribute__("compress_rom"),
+        )
+
+    def test_default_settings_strict(self):
+        try:
+            Settings.Settings({}, True)
+        except (TypeError, ValueError, BaseException) as e:
+            self.fail(f"Default settings no longer pass `validate_settings`\n{e}")
+
+    def test_world_count_min(self):
+        # TODO: The value checked in the assert should be constant to avoid needing to change this test for
+        #       intentional changes.
+        world_count_settings = Settings.Settings({"world_count": 0})
+        self.assertEqual(world_count_settings.world_count, 1)
+
+    def test_world_count_max(self):
+        # TODO: The value checked in the assert should be constant to avoid needing to change this test for
+        #       intentional changes.
+        world_count_settings = Settings.Settings({"world_count": 256})
+        self.assertEqual(world_count_settings.world_count, 255)
+
+    def test_get_numeric_seed_is_consistent(self):
+        test_settings = Settings.Settings({"seed": "unittest"})
+        duplicate_seed = Settings.Settings({})
+        duplicate_seed.seed = "unittest"
+
+        self.assertEqual(
+            test_settings.get_numeric_seed(),
+            duplicate_seed.get_numeric_seed(),
+            f"Using the same seed value resulted in different numeric seeds.",
+        )
+
+    def test_copy(self):
+        distribution_file = local_path(
+            Path("tests", "plando", "plando-beehives.json").__str__()
+        )
+        test_settings = Settings.Settings({})
+        test_settings.enable_distribution_file = True
+        test_settings.distribution_file = distribution_file
+        test_settings.load_distribution()
+
+        result_settings = test_settings.copy()
+
+        self.assert_settings_objects_are_equal(
+            result_settings, test_settings, False, {}
+        )
+
+    def test_update_with_default_settings_string(self):
```

Comment:
I'm not sure these tests are worth the added maintenance cost. Settings strings are going to change frequently.